### PR TITLE
Origin/editorialreview/zeebe

### PR DESCRIPTION
**Pull request by christinaausley**
Editorial review of Zeebe docs for grammar, punctuation, organization, and terminology.

### CODE REVIEW

File: docs/product-manuals/zeebe/deployment-guide/getting-started/deploy-a-process.md [hunk 1]
```diff
@@ -3,9 +3,11 @@ id: deploy-a-process
 title: "Deploy a process"
 ---
 
-_In this section, we're going to start up the Zeebe broker as well as Camunda Operate, a tool that gives you visibility into deployed processes and running process instances and contains tooling for fixing problems in those process instances._
+In this section, we'll start up the Zeebe broker and Camunda Operate—a tool that gives you visibility into deployed processes and running process instances, and contains tooling for fixing problems in those process instances.
```
Comment (pihme) [inline docs/product-manuals/zeebe/deployment-guide/getting-started/deploy-a-process.md]: :thought_balloon: With the new phrasing connection "and" it is less clear to me that Operate is the "tool that gives you visibility". Arguably, it maybe wasn't all that clear before.

So maybe something along the lines of: "In this section we'll start up the Zeebe broker - the underlying worklfow execution engine - and Operate - a tool that gives you..."

But then there might be too many hyphens. 

Also not sure why it's Zeebe broker but **Caumunda**  Operate since both are Camunda products. But that is more a marketing thing.

File: docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md [hunk 2]
```diff
@@ -18,21 +18,21 @@ If this is your first time working with Zeebe, we expect this entire guide to ta
...
 
-1.  [Camunda Modeler](https://docs.camunda.io/docs/product-manuals/modeler/camunda-modeler/install-the-modeler): A desktop modeling tool that we'll use to create and configure our process before we deploy it to Zeebe.
-1.  [Zeebe Distribution](https://github.com/camunda-cloud/zeebe/releases/tag/0.20.0): The Zeebe distribution contains the workflow engine where we'll deploy our process model; the engine is also responsible for managing the state of active process instances. Included in the distro is the Zeebe CLI, which we'll use throughout the tutorial. Please use Zeebe 0.20.0.
+1.  [Camunda Modeler](https://docs.camunda.io/docs/product-manuals/modeler/camunda-modeler/install-the-modeler): A desktop modeling tool we'll use to create and configure our process before we deploy it to Zeebe.
+1.  [Zeebe Distribution](https://github.com/camunda-cloud/zeebe/releases/tag/0.20.0): The Zeebe distribution contains the workflow engine where we'll deploy our process model; the engine is also responsible for managing the state of active process instances. Included in the distro is the Zeebe CLI, which we'll use throughout the tutorial. Use Zeebe 0.20.0.
```
Comment (pihme) [inline docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md]: **Mandatory:** Wow, we definitely need to change this. They should use the latest version always. Best would be a variable so it always updated with each documentation version. Applies both to the reference in the text, and to the link)

File: docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md [hunk 3]
```diff
@@ -18,21 +18,21 @@ If this is your first time working with Zeebe, we expect this entire guide to ta
...
 1.  [Camunda Operate](https://github.com/camunda-cloud/zeebe/releases/tag/0.20.0): An operations tool for monitoring and troubleshooting live process instances in Zeebe. Operate is currently available for free and unrestricted _non-production use_.
 1.  [Elasticsearch 6.8.0](https://www.elastic.co/downloads/past-releases/elasticsearch-6-8-0): An open-source distributed datastore that can connect to Zeebe to store process data for auditing, visualization, analysis, etc. Camunda Operate uses Elasticsearch as its underlying datastore, which is why you need to download Elasticsearch to complete this tutorial. Operate and Zeebe are compatible with Elasticsearch 6.8.0.
```
Comment (pihme) [inline docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md]: **Mandatory** This also needs to be changed to 7.x, but here I am not sure

@npepinpe Do we have a minimum version for elastic? How can we make sure that this part of the documentation stays up to date?

File: docs/product-manuals/zeebe/deployment-guide/operations/resource-planning.md [hunk 4]
```diff
@@ -75,72 +73,73 @@ Environment Variables
...
-- `state` - the active state. Deployed processes, active process instances, etc. Completed process instances or jobs are removed.
-- `snapshot` - a state at a certain point in time
+- `segments` - The data of the log split into segments. The log is only appended, and its data can be deleted when it becomes part of a new snapshot.
+- `state` - The active state. Deployed processes, active process instances, etc. Completed process instances or jobs are removed.
+- `snapshot` - A state at a certain point in time.
 
 > **Pitfalls**
 >
-> If you want to avoid exploding your disk space usage, here are a few pitfalls to avoid:
+> To avoid exploding your disk space usage, here are a few pitfalls to avoid:
```
Comment (pihme) [inline docs/product-manuals/zeebe/deployment-guide/operations/resource-planning.md]: I still find the word "exploding" strange. Does that make sense to you, as a native speaker?

**Review by pihme — CHANGES_REQUESTED** [review]
Review part 1

Some comments. Those marked *Mandatory* are mandatory changes, all others are optional

File: docs/product-manuals/zeebe/open-source/exporters.md [hunk 5]
```diff
@@ -47,87 +44,69 @@ all exporters interact in the same way with the broker, which is defined by the
...
-> **Note:** although Zeebe tries to reduce the amount of duplicate records an
-> exporter has to handle, it is likely that it will have to; therefore, it is
-> necessary that export operations be idempotent.
->
-> This can be implemented either in the exporter itself, but if it exports to an
-> external system, it is recommended that you perform deduplication there to reduce
-> the load on Zeebe itself. Refer to the exporter specific documentation for how
-> this is meant to be achieved.
+:::note
+Although Zeebe tries to reduce the amount of duplicate records an exporter has to handle, it is likely that it will have to. Therefore, it is necessary that export operations be idempotent. This can be implemented either in the exporter itself, but if it exports to an external system, it is recommended you perform deduplication there to reduce the load on Zeebe itself. Refer to the exporter-specific documentation for how this is meant to be achieved.
```
Comment (pihme) [inline docs/product-manuals/zeebe/open-source/exporters.md]: I find the second half of this sentence hard to understand: "Although Zeebe tries to reduce the amount of duplicate records an exporter has to handle, it is likely that it will have to."

**Review by pihme — COMMENTED** [review]
review part II (and finished)

More comments

File: docs/product-manuals/zeebe/deployment-guide/getting-started/deploy-a-process.md (hunk 1 above)
Comment (christinaausley) [reply docs/product-manuals/zeebe/deployment-guide/getting-started/deploy-a-process.md]: Agreed on this -- your proposed text is much more clear. Added into latest commit.

File: docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md (hunk 2 above)
Comment (christinaausley) [reply docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md]: Would you prefer to update and link to this: https://camunda.com/blog/2021/02/announcing-zeebe-operate-and-tasklist-1-0-0-alpha1/ ?

File: docs/product-manuals/zeebe/deployment-guide/operations/resource-planning.md (hunk 4 above)
Comment (christinaausley) [reply docs/product-manuals/zeebe/deployment-guide/operations/resource-planning.md]: This is a little awkward. How's this:

"To avoid exceeding your disk space, here are a few pitfalls to avoid"

File: docs/product-manuals/zeebe/open-source/exporters.md (hunk 5 above)
Comment (christinaausley) [reply docs/product-manuals/zeebe/open-source/exporters.md]: How's the following:

"Zeebe tries to reduce the amount of duplicate records an exporter handles, but it is helpful to ensure export operations are idempotent."

**Comment by christinaausley** [comment]
@pihme Thank you so much for your review and very helpful comments! I've proposed changes to resolve these issues in the comments.

**Comment by akeller** [comment]
@menski & @felix-mueller want to review all the deployment guides in the future, but I see there were some comments that came up in review for the Zeebe deployment guide.

My thoughts - if there are quick fixes to do today, let's do them. If there are more complicated changes (like how to reference new versions, what to link where, etc.) we can wait on this. I'd like to have minimal places to update with each new version and make use of pages like our Update Guides, Announcements, and other pages that *should* be updated with each version.

File: docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md (hunk 2 above)
Comment (pihme) [reply docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md]: I would probably rephrase it to "Use the latest release" and then link to this page: https://github.com/camunda-cloud/zeebe/releases
The blog article is already outdated. Right now it would be: https://camunda.com/blog/2021/07/camunda-cloud-110-released/

But in a couple of weeks it will be 1.1.2. 

The link to the github page is a stable one, and then user can pick out the latest release for their needs

File: docs/product-manuals/zeebe/open-source/exporters.md (hunk 5 above)
Comment (pihme) [reply docs/product-manuals/zeebe/open-source/exporters.md]: I feel both are not ideal. Exporters don't **have** to be idempotent. It is just a convenient shortcut to describe how they **should** operate.

Maybe we would be better served to describe the contract that an exporter has to follow.

Something like (tech jargon, not a suggestion how to phrase it): 
- exporters are called with at least once semantics for the records to be exported
- this means they might be called more than once for any given record
- deduplication can be performed based on the record key, which is unique

Also not sure what we mean by "exporter-specific documentation". I searched the documentation for a helpful section, but couldn't find it. If it is there, we should link to it. If not, I think we should remove this reference.

**Comment by saig0** [comment]
@pihme nothing to add from my side :rocket:

File: docs/product-manuals/zeebe/open-source/exporters.md (hunk 5 above)
Comment (christinaausley) [reply docs/product-manuals/zeebe/open-source/exporters.md]: Here's what I've updated to, does it make sense to you or are there pieces you would add/remove?

"It's best to reduce the amount of duplicate records an exporter handles alongside Zeebe. Exporters are called at least once for the records to export, which means they may be called more than once for any given record. We recommend performing deduplication based on the unique record key to reduce the load on Zeebe."

Comment (pihme) [reply docs/product-manuals/zeebe/open-source/exporters.md]: @saig0/@npepinpe : What do you think of the proposal? 

Personally, I am stumbling a bit over "reduce the load on Zeebe". I don't think that's the goal here, and also if deduplication is done badly it could hurt Zeebe performance. Plus, I think deduplication should rather be done on the backend.

At the same time, I feel getting it exactly right might be too complicated.

File: docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md (hunk 3 above)
Comment (npepinpe) [reply docs/product-manuals/zeebe/deployment-guide/getting-started/tutorial-setup.md]: 7.x is the minimum mandatory version with Zeebe 1.x. I'm not sure how we would keep it up to date in an automated way =/ To be fair, the next time this needs to be done is with major versions, so should not happen very often. 

Generally, the tutorial would need to be updated, no? It still points to Zeebe 0.20.0, which is very outdated at this point, but I imagine this falls outside the scope of this PR?

File: docs/product-manuals/zeebe/open-source/exporters.md (hunk 5 above)
Comment (npepinpe) [reply docs/product-manuals/zeebe/open-source/exporters.md]: You're right that reducing the load on Zeebe is not the goal of de-duplication. Further, this de-duplication should occur on your target system, not in Zeebe (in this sense, here it _is_ to reduce the load on Zeebe :smile:). 

So the important bits:

- Zeebe exports with at least once semantics, so you _will_ have to deal with duplicates
- De-duplication can be performed based on the position and partition ID - not all records have keys, but every record has a partition ID, and a unique position relative to that partition.

We can drop the reduce load part, and focus on communicating that it's crucial for the backend system to deal with duplicates. We could keep something to the effect of trying to do as little as possible in the exporter to reduce the load on the broker though. A note also on not blocking would be good, since blocking means blocking on the actor threads entirely, which can have a big impact on the system.

The exporter specific documentation here means the documentation for a specific exporter, e.g. Elasticsearch Exporter documentation, or the Kafka exporter documentation, etc. It's also possible some parts of our docs don't make sense in the new structure as well.

**Review by npepinpe — APPROVED** [review]
Nothing to add on my side - I think there's some outdated things (i.e. the tutorial on 0.20.0 for example) that were raised already, and there may be more, but (and correct me if I'm wrong) I think those were outside the scope of this PR.

File: docs/components/zeebe/open-source/exporters.md [hunk 6]
```diff
@@ -94,7 +94,7 @@ set by the exporter itself once it can guarantee a record has been successfully
 updated.
 
 :::note
-Zeebe exports with at least once semantics, so you will have to deal with duplicates. It's best to reduce the amount of duplicate records an exporter handles alongside Zeebe, doing as little as possible in the exporter to reduce the load on the broker. We recommend performing deduplication based on the unique record key. We also recommend refraining from blocking, given blocking on the actor threads entirely, which may have a large impact on the system.
+Zeebe exports with at least once semantics, so you will have to deal with duplicates. It's best to reduce the amount of duplicate records an exporter handles alongside Zeebe, doing as little as possible in the exporter to reduce the load on the broker. We recommend performing deduplication in your target system. Deduplication can be performed based on the position and partition ID. Not all records have keys, but every record has a partition ID, and a unique position relative to that partition. We also recommend refraining from blocking, given blocking on the actor threads entirely, which may have a large impact on the system.
```
Comment (pihme) [inline docs/components/zeebe/open-source/exporters.md:97]: Suggestion: 

Zeebe exports with at least once semantics, so you will have to deal with duplicates. It's best to reduce the amount of duplicate records an exporter handles alongside Zeebe, doing as little as possible in the exporter to reduce the load on the broker. We recommend performing deduplication in your target system. Deduplication can be performed based on the partition ID and record index. Not all records have keys, but every record has a partition ID, and a unique position relative to that partition. We also recommend refraining from blocking operations in the exporter, as it will have a negative impact on performance of the system.

Comment (npepinpe) [reply docs/components/zeebe/open-source/exporters.md:97]: Sounds good, but I would keep using the word record position and not record index to keep the language consistent across the whole documentation (including the Javadoc).

Comment (christinaausley) [reply docs/components/zeebe/open-source/exporters.md:97]: Adjusted and resolved, thanks!
